### PR TITLE
Add back-end endpoint for buying units

### PR DESCRIPTION
**Pull request by tomekzaw**
https://jira.lab.ki.agh.edu.pl/browse/CG-31
pair programming with @matix522

### CODE REVIEW

File: mechanics/mechanics.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
+from random import randint
 from typing import Sequence
 
 from model.Unit import Unit
 
+sqrt2 = 2 ** 0.5
+golden = (1 + 5 ** 0.5) / 2
```
Comment (matix522) [inline mechanics/mechanics.py:5]: should be golden_ration

File: mechanics/mechanics.py [hunk 2]
```diff
@@ -2,9 +2,8 @@
 
 from model.Unit import Unit
 
-sqrt2 = 2 ** 0.5
 golden = (1 + 5 ** 0.5) / 2
 
 
 def generate_unit(round: int) -> Unit:
-    return Unit(price=randint(int(sqrt2 ** round), int(3 * sqrt2 ** round)) * 100)
+    return Unit(price=randint(int(1.5 ** round), int(3 * 1.5 ** round)) * 100)
```
Comment (matix522) [inline mechanics/mechanics.py]: Magic numbers

File: mechanics/mechanics.py [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
 
 def generate_unit(round: int) -> Unit:
-    return Unit(price=randint(int(1.5 ** round), int(3 * 1.5 ** round)) * 100)
+    # profession = choice(['magician', 'dps', 'tank', 'sniper'])
+    profession = 'sniper'
+
+    def gen(min, max, multiplier: float = 1.0) -> int:
+        return int(uniform(min, max) * multiplier ** (round - 1))
+
+    if profession == 'sniper':
```
Comment (matix522) [inline mechanics/mechanics.py:15]: A lot of magic numbers, but I think they are not final.

File: mechanics/mechanics.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from random import choice, uniform
+
+from model.Unit import Unit
+
+golden = (1 + 5 ** 0.5) / 2
+
+
+def generate_unit(round: int) -> Unit:
+    # profession = choice(['magician', 'dps', 'tank', 'sniper'])
```
Comment (matix522) [inline mechanics/mechanics.py:9]: commented code.

Comment (matix522) [reply mechanics/mechanics.py:9]: leftover comments

File: mechanics/mechanics.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    profession = 'sniper'
+
+    def gen(min, max, multiplier: float = 1.0) -> int:
+        return int(uniform(min, max) * multiplier ** (round - 1))
+
+    if profession == 'sniper':
```
Comment (matix522) [inline mechanics/mechanics.py:15]: Always true

File: mechanics/mechanics.py (hunk 1 above)
Comment (tomekzaw) [reply mechanics/mechanics.py:5]: fixed

File: mechanics/mechanics.py (hunk 2 above)
Comment (tomekzaw) [reply mechanics/mechanics.py]: obsolete

File: mechanics/mechanics.py (hunk 3 above)
Comment (tomekzaw) [reply mechanics/mechanics.py:15]: yes, they are to be tuned yet

File: mechanics/mechanics.py (hunk 5 above)
Comment (tomekzaw) [reply mechanics/mechanics.py:15]: profession will be handled soon

File: mechanics/mechanics.py (hunk 4 above)
Comment (tomekzaw) [reply mechanics/mechanics.py:9]: profession choice will be implemented very soon

**Pull request merged**